### PR TITLE
Handle gem2s response

### PR DESCRIPTION
**Pull request by cosa65**
# Background
#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-947
https://biomage.atlassian.net/browse/BIOMAGE-984

#### Link to staging deployment URL 
https://ui-martinfosco-ui274-api127-w.scp-staging.biomage.net/

#### Links to any Pull Requests related to this
https://github.com/biomage-ltd/ui/pull/274
https://github.com/biomage-ltd/pipeline/pull/67

#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [ ] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: src/api/general-services/pipeline-status.js [hunk 1]
```diff
@@ -102,8 +105,12 @@ const getStepsFromExecutionHistory = (events) => {
      *  - a step is only considered completed if it has been completed for all iteration of the Map
      *  - steps are returned in the completion order, and are unique in the returned array
      */
-const getPipelineStatus = async (experimentId) => {
```
Comment (xverges) [inline src/api/general-services/pipeline-status.js:105]: Could you delete my misleading comments in the first two lines after "The code assumes"? They have not been true for a while!

File: src/utils/schema-validator.js [hunk 2]
```diff
@@ -11,10 +11,10 @@ const logger = require('./logging');
 const validateRequest = async (request, schemaPath) => {
   const specPath = path.resolve(__dirname, '..', 'specs', 'models', schemaPath);
 
-  console.log('received validation request');
-  console.log({ request });
-  console.log({ schemaPath });
-  console.log({ specPath });
+  logger.log('received validation request');
```
Comment (xverges) [inline src/utils/schema-validator.js]: Isn't this too noisy? I am under the impression that we are carrying on some debug statements that will clutter our logs...

File: tests/api/general-services/pipeline-status.test.js [hunk 3]
```diff
@@ -337,9 +338,15 @@ describe('getStepsFromExecutionHistory', () => {
 });
 
 jest.mock('../../../src/api/route-services/experiment', () => jest.fn().mockImplementation(() => ({
-  getPipelineHandle: () => ({
```
Comment (xverges) [inline tests/api/general-services/pipeline-status.test.js:340]: The tests are failing: `The module factory of jest.mock() is not allowed to reference any out-of-scope variables`. I think that if you move this code to a `beforeAll` call inside the `describe` function, we will be good

File: src/api/routes/pipelines.js [hunk 4]
```diff
@@ -17,14 +17,13 @@ module.exports = {
     },
   ],
   'pipelines#create': [
-    expressAuthorizationMiddleware,
```
Comment (xverges) [inline src/api/routes/pipelines.js]: Why don't we need authorization here?

Comment (cosa65) [reply src/api/routes/pipelines.js]: Good quesiton, restored it

**Review by marcellp — APPROVED** [review]

**Pull request merged**